app/models/date_parser.rb

```diff
@@ -22,6 +22,7 @@ def date
 
     return nil if date_values.all?(&:blank?)
     return struct_from_hash if date_values.any?(&:blank?)
+    return struct_from_hash if date_values.any?(&:negative?)
```

Comment:
Think we only need to check whether the month or day is negative, not the year, as `Date.new(-2020,1,1)` is valid.

```suggestion
    return struct_from_hash if date_values[1].negative? || date_values[2].negative?
```